Fix 'node-init' in GKE's 'cos' images.

bmcustodio opened the pull request:
Turns out that in GKE's `cos` images the `containerd` binary is still present even though `/etc/containerd/config.toml` is not. Hence, the kubelet wrapper would still be installed for these images according to the current check, even though it's not necessary. What's worse, starting the kubelet would fail because the `sed` command targeting the aforementioned file would fail.

This PR changes the check to rely on the presence of the `--container-runtime-endpoint` flag in the kubelet, which is probably a more reliable way of detecting `*_containerd` flavours and only applying the fix in these cases.

Fixes #19015.

```release-note
Fix 'node-init' in GKE's 'cos' images.
```

kaworu commented on install/kubernetes/cilium/files/nodeinit/startup.bash:
```diff
@@ -22,9 +22,11 @@ fi
 GKE_KUBERNETES_BIN_DIR="/home/kubernetes/bin"
-if [[ -f "${GKE_KUBERNETES_BIN_DIR}/gke" ]] && command -v containerd &>/dev/null; then
+KUBELET_DEFAULTS_FILE="/etc/default/kubelet"
+if [[ -f "${GKE_KUBERNETES_BIN_DIR}/gke" ]] && [[ $(grep -cE '\--container-runtime-endpoint' "${KUBELET_DEFAULTS_FILE}") == "1" ]]; then
```
If I'm reading this correctly we're looking for exactly `--container-runtime-endpoint` and if correct then I'm not sure why we need `-E` here:

```suggestion
if [[ -f "${GKE_KUBERNETES_BIN_DIR}/gke" ]] && [[ $(grep -cF -- '--container-runtime-endpoint' "${KUBELET_DEFAULTS_FILE}") == "1" ]]; then
```

bmcustodio committed:
Fix 'node-init' in GKE's 'cos' images.

Turns out that in GKE's 'cos' images the 'containerd' binary is still
present even though '/etc/containerd/config.toml' is not. Hence, the
kubelet wrapper would still be installed for these images according to
the current check, even though it's not necessary. What's worse,
starting the kubelet would fail because the 'sed' command targeting the
aforementioned file would fail.

This PR changes the check to rely on the presence of the
'--container-runtime-endpoint' flag in the kubelet, which is probably a
more reliable way of detecting '*_containerd' flavours and only applying
the fix in these cases.

Fixes #19015.

Signed-off-by: Bruno M. Custódio <brunomcustodio@gmail.com>
Co-authored-by: Alexandre Perrin <alex@kaworu.ch>
